presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -118,6 +119,15 @@ fun NavigationGraph(
                 )
             }
 
+            /** TODO Once the insertion has completed the state will change to true then
+             *  we will know that the insertion has completed and we can popbackstack
+             *  to go back to the agenda screen */
+            LaunchedEffect(key1 = eventScreenState.hasInsertedCompleted) {
```

Comment:
Is this correct?

Reply:
Yes

Reply:
@MKiperszmid I used to pass in the ViewModel but since @philipplackner has taught as a different method of passing in the state and the events I prefer to go with @philipplackner approach.

Reply:
For me it's a no brainer to keep composables ViewModel-free, just brings advantages with no disadvantages and no extra effort 🤔